bd-filters/src/lib.rs

```diff
@@ -0,0 +1,208 @@
+    Ok(match transform_type {
+      Transform_type::CaptureFields(config) => {
+        Self::CaptureField(CaptureField::new(config).map_err(|e| e.context("CaptureFields"))?)
+      },
+      Transform_type::SetField(config) => {
+        Self::SetField(SetField::new(config).map_err(|e| e.context("SetField"))?)
+      },
+    })
+  }
+
+  fn apply(&self, log: &mut Log) {
+    match self {
+      Self::CaptureField(capture_field) => capture_field.apply(log),
+      Self::SetField(set_field) => set_field.apply(log),
+    }
+  }
+}
+
+//
+// CaptureField
+//
+
+struct CaptureField {
+  field_name: String,
+}
+
+impl CaptureField {
+  fn new(config: filter::transform::CaptureFields) -> Result<Self> {
+    let Some(ref fields_type) = config.fields.fields_type else {
+      anyhow::bail!("no fields_type")
+    };
+
+    let Fields_type::Single(field) = fields_type;
+
+    Ok(Self {
+      field_name: field.name.clone(),
+    })
+  }
+
+  fn apply(&self, log: &mut Log) {
+    // Look for a field that's supposed to be captured.
+    let Some(field_position) = log
+      .matching_fields
+      .iter()
+      .position(|field| field.key == self.field_name)
+    else {
+      // Matching field with a given key doesn't exist so there is nothing to capture.
+      return;
+    };
+
+    let field = log.matching_fields.remove(field_position);
+
+    // If a field that's supposed to be captured exists, remove it from the list of captured fields
+    // before adding it.
+    log.fields.retain(|field| field.key != self.field_name);
+    log.fields.push(field)
+  }
+}
+
+//
+// FieldType
+//
+
+enum FieldType {
+  MatchingOnly,
+  Captured,
+}
+
+//
+// SetField
+//
+
+struct SetField {
+  field_name: String,
+  value: String,
+  field_type: FieldType,
+}
+
+impl SetField {
+  fn new(config: filter::transform::SetField) -> Result<Self> {
+    let field_type = config
+      .field_type
+      .enum_value()
+      .map_err(|e| anyhow!(format!("invalid enum value {e}")))?;
+
+    Ok(Self {
+      field_name: config.name,
+      value: config.value,
+      field_type: match field_type {
+        filter::transform::set_field::FieldType::UNKNOWN => anyhow::bail!("unknown field_type"),
+        filter::transform::set_field::FieldType::CAPTURED => FieldType::Captured,
+        filter::transform::set_field::FieldType::MATCHING_ONLY => FieldType::MatchingOnly,
+      },
+    })
+  }
+
+  fn apply(&self, log: &mut Log) {
+    let field = bd_log_primitives::LogField {
+      key: self.field_name.clone(),
+      value: self.value.clone().into(),
+    };
+
+    match self.field_type {
+      FieldType::MatchingOnly => {
+        log
+          .matching_fields
+          .retain(|field| field.key != self.field_name);
+        log.matching_fields.push(field)
+      },
+      FieldType::Captured => {
+        log.fields.retain(|field| field.key != self.field_name);
+        log.fields.push(field)
```

Comment:
not a big deal but you could add some "override_field" helper since this pattern seems to repeat a bit